### PR TITLE
setup for sqllite3

### PR DESCRIPTION
**Pull request by oneEyedSunday**
For https://github.com/adelowo/sdump/issues/33

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -77,3 +84,11 @@ type Config struct {
 	TUI      TUIConfig  `mapstructure:"tui" json:"tui,omitempty" yaml:"tui"`
 	Cron     CronConfig `mapstructure:"cron" yaml:"cron" json:"cron,omitempty"`
 }
+
+func (cfg *Config) GetDatabase() (*bun.DB, error) {
+	if cfg.HTTP.Database.Driver == "sqllite" {
```
Comment (adelowo) [inline config/config.go]: `sqlite` instead

File: datastore/sqlite/sqlite.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+func New(dsn string, logQueries bool) (*bun.DB, error) {
+	sqlite, err := sql.Open(sqliteshim.ShimName, dsn)
+	if err != nil {
+		panic(err)
+	}
+
+	db := bun.NewDB(sqlite, sqlitedialect.New())
+
+	if logQueries {
+		db.AddQueryHook(bundebug.NewQueryHook(bundebug.WithVerbose(true)))
```
Comment (adelowo) [inline datastore/sqlite/sqlite.go]: Please let us add https://github.com/oiime/logrusbun instead so the log format stays consistent. Always wanted to do this but never had the time

**Review by adelowo — CHANGES_REQUESTED** [review]
Looks good to me asides the suggested Changes

File: config/config.go [hunk 3]
```diff
@@ -34,6 +40,7 @@ type HTTPConfig struct {
 	Database struct {
 		DSN        string `mapstructure:"dsn" json:"dsn,omitempty" yaml:"dsn"`
 		LogQueries bool   `mapstructure:"log_queries" json:"log_queries,omitempty" yaml:"log_queries"`
+		Driver     string `mapstructure:"driver" json:"driver,omitempty" yaml:"driver,omitempty"`
```
Comment (adelowo) [inline config/config.go]: Please define enums for this. So people can know the supported driver types

File: config/config.go [hunk 4]
```diff
@@ -77,3 +84,11 @@ type Config struct {
...
 	Cron     CronConfig `mapstructure:"cron" yaml:"cron" json:"cron,omitempty"`
 }
+
+func (cfg *Config) GetDatabase() (*bun.DB, error) {
+	if cfg.HTTP.Database.Driver == "sqllite" {
+		return sqlite.New(cfg.HTTP.Database.DSN, cfg.HTTP.Database.LogQueries)
+	}
+
+	return postgres.New(cfg.HTTP.Database.DSN, cfg.HTTP.Database.LogQueries)
+}
```
Comment (adelowo) [inline config/config.go]: also can we encapsulate this into just one `New` function and depending on the driver, we return the correctly configured one. having Postgres and sqlite packages just to export `New` seems overkill to me

**Comment by adelowo** [comment]
aside these, I think I am happy to merge this.

**Comment by oneEyedSunday** [comment]
> aside these, I think I am happy to merge this.

Hey man. I'll get to this in the next 24

**Comment by oneEyedSunday** [comment]
> aside these, I think I am happy to merge this.

requested changes added.

**Comment by adelowo** [comment]
> > aside these, I think I am happy to merge this.
> 
> requested changes added.

thanks a lot for the work! Greatly appreciated!

**Review by adelowo — APPROVED** [review]

**Pull request merged**